flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -547,7 +552,7 @@ class TaskManager(
 
         // successful registration. associate with the JobManager
         // we disambiguate duplicate or erroneous messages, to simplify debugging
-        case AcknowledgeRegistration(_, leaderSessionID, jobManager, id, blobPort) =>
+        case AcknowledgeRegistration(_, leaderSessionID, jobManager, id, blobPort, jobManagerID) =>
```

Comment:
But the initial registration must have happened with `AcknowledgeRegistration`, right? Correct me if I'm wrong.